articles/express/index.md

```diff
@@ -0,0 +1,360 @@
+
+We are using the [path module](https://nodejs.org/api/path.html) to join the relative URL (from the request) and the directory name. 
+
+The [fs module](https://nodejs.org/api/fs.html) provides an API for interacting with the file system. We are checking if the file exists, if it does not, we will go to next function in the stack if it does we will return that file using `res.sendFile`.
+
+
+#### Using Third-Party Middleware
+We can write our own middleware functions or import them similar to how we imported our modules in Node.js using `require`. 
+
+Let's use a popular open-source logger called [morgan](http://expressjs.com/en/resources/middleware/morgan.html) instead of writing our own logging function.
+
+Install it using npm.
+
+```bash
+$ npm install morgan
+```
+
+We can call the `use()` on the Express app object to add the middleware to the stack.
+
+```JavaScript
+
+var express = require("express");
+var logger_morgan = require("morgan");
+
+var app = express();
+...
+app.use(logger_morgan("short")); // logs short notation of requests
+...
+```
+
+Express comes with **`express.static`** middleware bundled with it, it can be used to serve static files instead of the function in the previous section. It provides better security and performance than the function we wrote.
+
+```JavaScript
+app.use(express.static("static") //relative path
+```
+
+Any requested files in the directory "static" are served. `localhost:3000/dummy_file.txt` will show the same result as above. We can call `static()` multiple times. 
+
+If the file is not in the one directory, then the request is passed onto the next `static()` call. When your app gets a request like `/hello.html`, it returns the hello.html file in the public directory.
+
+### Routing
+Express makes request handling easier by mapping requests to different request handlers. A request handler is a function which handles all the requests to a specific path with a specific HTTP method. 
+
+In the basic example above, we saw how to handle a `GET` request. As an application grows in size the routes grow as well as do the request handlers.
+
+Lets see how we can use [Routers](http://expressjs.com/en/4x/api.html#router) to split a large app into smaller, maintainable functions. 
+
+According to the documentation, a Router is "an isolated instance of middleware and routes. Routers can be thought of as “mini” applications only capable of performing middleware and routing".
+
+Routers can be used like middleware functions, they can be "`.use()`d". A simple example:
+
+```JavaScript
+//app.js the main file
+
+var express = require("express");
+var apiRouter = require("./routes/api_router");
+
+var app = express();
+
+app.use("/api", apiRouter);
+
+app.listen(3000);
+```
+
+```JavaScript
+//routes/api_router.js
```

Comment:
Please specify to the reader they have to create the routes folder and api_router.js file because they may gloss over the comments.